src/main/java/minijava/ir/optimize/ConstantFolder.java

```diff
@@ -185,4 +185,33 @@ public void visit(Shrs node) {
   public void visit(Sub node) {
     visitBinaryOperation((lhs, rhs) -> lhs.sub(rhs), node, node.getLeft(), node.getRight());
   }
+
+  @Override
+  public void visit(Phi node) {
+    int n = node.getPredCount();
```

Comment:
I don't understand: Why does constant folding need to change when implementing control flow optimization???